### PR TITLE
dpcmd: Include the config file for the program to work

### PR DESCRIPTION
**Pull request by superjeng1**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This file is required for the program to actually work, so I added it in.

On macOS, it would require this [pull request](https://github.com/DediProgSW/SF100Linux/pull/83) to work. It was already merged, but the maintainer has not made a release since then. But this should still be added in.

### CODE REVIEW

**Comment by superjeng1** [comment]
Oh, the maintainer had just created a new release, should I update it in this same pull request? Or should I create another one for it?

File: Formula/d/dpcmd.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ class Dpcmd < Formula
   def install
     system "make"
     bin.install "dpcmd"
+    include.install "ChipInfoDb.dedicfg"
```
Comment (carlocab) [inline Formula/d/dpcmd.rb:30]: Can we add a test that fails when this program is missing?

Comment (superjeng1) [reply Formula/d/dpcmd.rb:30]: The test does fail, but only if the programmer (a usb device) was attached unfortunately. I haven't found a way to make it show an error without it attached.

Comment (superjeng1) [reply Formula/d/dpcmd.rb:30]: Oh sorry, actually, the current test won't fail even if the device is connected.

```
> dpcmd -rSTDOUT -a0x100 -l0x23

DpCmd Linux 1.14.20.01 Engine Version:
Last Built on Jan 11 2024





> dpcmd --list

DpCmd Linux 1.14.20.01 Engine Version:
Last Built on Jan 11 2024


Error: chip not identified or chip is not supported.


```

Comment (superjeng1) [reply Formula/d/dpcmd.rb:30]: The exit code for the first one was `1` and the second one was `10`.

**Comment by carlocab** [comment]
> Oh, the maintainer had just created a new release, should I update it in this same pull request? Or should I create another one for it?

Just update it here, I think. You can just amend your existing commit for it.

File: Formula/d/dpcmd.rb (hunk 1 above)
Comment (carlocab) [inline Formula/d/dpcmd.rb:30]: Maybe I'm misreading https://github.com/DediProgSW/SF100Linux/pull/83, but it seems like this should be where we're installing this?
```suggestion
    (share/"DediProg").install "ChipInfoDb.dedicfg"
```

Comment (superjeng1) [reply Formula/d/dpcmd.rb:30]: Oh sorry, that is true. I used another patch for the one in my tap.

Comment (superjeng1) [reply Formula/d/dpcmd.rb]: Ok that was fixed. Thanks for pointing that out. I have tested the build fresh again and it did work as intended.

```
> HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source dpcmd
==> Auto-updating Homebrew...
Adjust how often this is run with HOMEBREW_AUTO_UPDATE_SECS or disable with
HOMEBREW_NO_AUTO_UPDATE. Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Warning: No remote 'origin' in /opt/homebrew/Library/Taps/superjeng1/homebrew-tap, skipping update!
==> You have HOMEBREW_NO_INSTALL_FROM_API set
Homebrew >=4.1.0 is dramatically faster and less error-prone when installing
from the JSON API. Please consider unsetting HOMEBREW_NO_INSTALL_FROM_API.
This message will only be printed once.


==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Tapping homebrew/cask
Cloning into '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 1112166, done.
remote: Counting objects: 100% (168607/168607), done.
remote: Compressing objects: 100% (561/561), done.
remote: Total 1112166 (delta 168148), reused 168046 (delta 168046), pack-reused 943559 (from 1)
Receiving objects: 100% (1112166/1112166), 322.21 MiB | 6.95 MiB/s, done.
Resolving deltas: 100% (831065/831065), done.
Tapped 2 commands and 7169 casks (7,240 files, 358.5MB).

You have 56 outdated formulae and 1 outdated cask installed.

==> Fetching dpcmd
==> Downloading https://github.com/DediProgSW/SF100Linux/archive/refs/tags/V1.14.21,x.tar.gz
==> Downloading from https://codeload.github.com/DediProgSW/SF100Linux/tar.gz/refs/tags/V1.14.21%2Cx
######################################################################################################################################################################################################################################################### 100.0%
==> make
🍺  /opt/homebrew/Cellar/dpcmd/1.14.21: 7 files, 846.6KB, built in 1 second
==> Running `brew cleanup dpcmd`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
Removing: /Users/willyhsiao/Library/Caches/Homebrew/dpcmd_bottle_manifest--1.14.20... (12.4KB)
Removing: /Users/willyhsiao/Library/Caches/Homebrew/dpcmd--1.14.20... (56.1KB)
Removing: /Users/willyhsiao/Library/Caches/Homebrew/dpcmd--1.14.20.tar.gz... (110.4KB)

> dpcmd --list

DpCmd macOS 1.14.21.12 Engine Version:
Last Built on Nov 14 2024


Alverstone-128M(Top)		by Micron(Numonyx)
Alverstone-128M(Bottom)		by Micron(Numonyx)
P5Q-128M		by Micron(Numonyx)
M25P05A		by Micron(Numonyx)
M25P05VP		by Micron(Numonyx)
M25P10A		by Micron(Numonyx)
[.....]
```

**Comment by superjeng1** [comment]
When would the bottles be rebuilt? Do I have to do anything?

**Comment by daeho-ro** [comment]
@superjeng1 It will be done automatically after the review :)

**Comment by superjeng1** [comment]
Can someone please review the changes? Thanks!

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by superjeng1** [comment]
Looks like all tests has passed, please go ahead with the merge. Thanks!

**Pull request merged**